src/lib/alloc.c

```diff
@@ -647,14 +647,22 @@ static void alloc_trace_heap(enum mem_zone zone, uint32_t caps, size_t bytes)
 
 #define DEBUG_TRACE_PTR(ptr, bytes, zone, caps, flags) \
 	do { \
-		if (!ptr) { \
-			tr_err(&mem_tr, "failed to alloc 0x%x bytes zone 0x%x caps 0x%x flags 0x%x", \
+		alloc_trace_heap(zone, caps, bytes); \
```

Comment:
We only need to dump this when the allocation fails otherwise it will flood the trace.

Reply:
my logic here is:
a. always trace errors when the allocation failed.
b. trace the updated heap status for the successful allocation, only when CONFIG_DEBUG_HEAP is selected ('no' by default) and the DEBUG_TRACE_PTR() is called. It will somewhat 'flood' the trace, but that's what our intention of 'CONFIG_DEBUG_HEAP'?